src/Core/PluginRegistry.php

```diff
@@ -0,0 +1,194 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright 2005-2011, Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.6.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Core;
+
+use Cake\Event\Event;
+use Cake\Event\EventManager;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use RuntimeException;
+
+/**
+ * Plugin Registry
+ */
+class PluginRegistry extends ObjectRegistry implements PluginRegistryInterface
+{
+
+    /**
+     * @inheritDoc
+     */
+    public function load($objectName, $config = [])
+    {
+        $plugin = parent::load($objectName, $config);
+
+        if (class_exists('\Cake\Event\Event')) {
+            $event = new Event('Plugin.loaded', $this, [
+                'name' => $objectName,
+                'plugin' => $plugin,
+                'config' => $config
+            ]);
+            EventManager::instance()->dispatch($event);
```

Comment:
Once #11692 is done the application event manager instance could be injected into plugin registry instead of fetching the global instance through static method.

Reply:
This might become a little off topic but by looking at this PR and the use case(s), I think we begin to use it like a DI container? Wouldn't it be better to wait for that?

Reply:
I think we can separate off events from this. #11692 will at solve several event manager related requirements.